src/GitVersionTfsTask/Tasks/GitVersion/GitVersion.ts

```diff
@@ -0,0 +1,99 @@
+import tl = require('vsts-task-lib/task');
+import path = require('path');
+import fs = require('fs-extra');
+import q = require('q');
+import XRegExp = require('xregexp');
+
+var variables = [
+    "GitVersion_Major",
+    "GitVersion_Minor",
+    "GitVersion_Patch",
+    "GitVersion_PreReleaseTag",
+    "GitVersion_PreReleaseTagWithDash",
+    "GitVersion_PreReleaseLabel",
+    "GitVersion_PreReleaseNumber",
+    "GitVersion_BuildMetaData",
+    "GitVersion_BuildMetaDataPadded",
+    "GitVersion_FullBuildMetaData",
+    "GitVersion_MajorMinorPatch",
+    "GitVersion_SemVer",
+    "GitVersion_LegacySemVer",
+    "GitVersion_LegacySemVerPadded",
+    "GitVersion_AssemblySemVer",
+    "GitVersion_FullSemVer",
+    "GitVersion_InformationalVersion",
+    "GitVersion_BranchName",
+    "GitVersion_Sha",
+    "GitVersion_NuGetVersionV2",
+    "GitVersion_NuGetVersion",
+    "GitVersion_NuGetPreReleaseTagV2",
+    "GitVersion_NuGetPreReleaseTag",
+    "GitVersion_CommitsSinceVersionSource",
+    "GitVersion_CommitsSinceVersionSourcePadded",
+    "GitVersion_CommitDate"
+]
+
+var updateAssemblyInfo = tl.getBoolInput('updateAssemblyInfo');
+var updateAssemblyInfoFilename = tl.getInput('updateAssemblyInfoFilename');
+var additionalArguments = tl.getInput('additionalArguments');
+var gitVersionPath = tl.getInput('gitVersionPath');
+
+var currentDirectory = __dirname;
+
+var sourcesDirectory = tl.getVariable("Build.SourcesDirectory")
+
+if (!gitVersionPath) {
+    gitVersionPath = tl.which("GitVersion.exe");
+    gitVersionPath = path.join(currentDirectory, "GitVersion.exe");
+}
+
```

Comment:
The process would be try to find gitversion.exe in path. If not found use bundled version. I think adding a advanced option `Always use bundled version` could be nice too.